Vision/synth.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This script is used to synthesize generated parts of this library."""
+
+import synthtool as s
+import synthtool.gcp as gcp
+import logging
+
+logging.basicConfig(level=logging.DEBUG)
+
+gapic = gcp.GAPICGenerator()
+common = gcp.CommonTemplates()
+
+# tasks has two product names, and a poorly named artman yaml
+v1_library = gapic._generate_code(
+    'vision', 'v1', 'php',
+    config_path='artman_vision_v1.yaml',
+    artman_output_name='google-cloud-vision-v1')
+
+# copy all src except ImageAnnotatorClient.php
+s.move(v1_library / f'src/V1/Gapic')
+s.move(v1_library / f'src/V1/resources')
+
+# copy proto files to src also
+s.move(v1_library / f'proto/src/Google/Cloud/Vision/', f'src/')
```

Comment:
There are also some metadata files related to protobuf. They need to end up here: https://github.com/GoogleCloudPlatform/google-cloud-php/tree/master/Vision/metadata/V1

Outside of that, looks good to me! Looking forward to running the tool.